ArduPlane/soaring.cpp

```diff
@@ -13,7 +13,8 @@ void Plane::update_soaring() {
     // status on active state changes.
     plane.g2.soaring_controller.update_active_state();
 
-    if (!g2.soaring_controller.is_active()) {
+    if (!g2.soaring_controller.is_active() || mission.get_in_landing_sequence_flag()) {
```

Comment:
Should the soaring controller be active at all?  It itself could look at the mission state.

Reply:
This is the main entry point, so returning at this stage means the soaring controller does nothing even though it's "active". "Active" just means the switches and parameters are suitable to take action.

The soaring controller currently doesn't look at the mission. My thinking was that it's better to avoid introducing that coupling for the sake of this feature.